it/v2/connectors/PersistenceConnectorSpec.scala

```diff
@@ -921,6 +922,178 @@ class PersistenceConnectorSpec
     }
   }
 
+  "GET /traders/:eori/movements/arrivals/:arrivalId/messages" - {
+
+    implicit val hc = HeaderCarrier()
+
+    def targetUrl(eoriNumber: EORINumber, arrivalId: MovementId) =
+      s"/transit-movements/traders/${eoriNumber.value}/movements/arrivals/${arrivalId.value}/messages/"
+
+    def targetUrlWithTime(eoriNumber: EORINumber, arrivalId: MovementId, receivedSince: OffsetDateTime) =
+      s"/transit-movements/traders/${eoriNumber.value}/movements/arrivals/${arrivalId.value}/messages/?receivedSince=${URLEncoder.encode(
+        DateTimeFormatter.ISO_OFFSET_DATE_TIME
+          .format(receivedSince),
+        StandardCharsets.UTF_8.name()
+      )}"
+    "on successful return of message IDs when no filtering is applied, return a success" in forAll(
+      arbitrary[EORINumber],
+      arbitrary[MovementId],
+      Gen.nonEmptyListOf(arbitrary[MessageSummary])
+    ) {
+      (eori, arrivalId, messageResponse) =>
+        server.stubFor(
+          get(
+            urlEqualTo(targetUrl(eori, arrivalId))
+          ).willReturn(
+            aResponse()
+              .withStatus(OK)
+              .withBody(
+                Json.toJson(messageResponse).toString()
+              )
+          )
+        )
+        val result = persistenceConnector.getArrivalMessageIds(eori, arrivalId, None)
+        whenReady(result) {
+          _ mustBe messageResponse
+        }
+    }
+
+    "on successful return of arrival message IDs when filtering by received date is applied, return a success" in forAll(
+      arbitrary[EORINumber],
+      arbitrary[MovementId],
+      OffsetDateTime.now(ZoneOffset.UTC),
+      Gen.nonEmptyListOf(arbitrary[MessageSummary])
+    ) {
+      (eori, arrivalId, time, messageResponse) =>
+        server.stubFor(
+          get(
+            urlEqualTo(targetUrlWithTime(eori, arrivalId, time))
+          )
```

Comment:
I just came to the realisation (while washing up, of all things) what you're doing wrong here. You've used `urlEqualTo`, but in fact, you want `urlPathEqualTo` in combination with `withQueryParam`

```suggestion
        server.stubFor(
          get(
            urlPathEqualTo(targetUrl(eori, arrivalId))
          )
            .withQueryParam("receivedSince", equalTo(DateTimeFormatter.ISO_DATE_TIME.format(time)))
```

Note that this is an internal call where we explicity make a call in UTC. I'd forgotten that little bit when talking through this, BUT your test was wrong anyway because you were formatting using the `ISO_OFFSET_DATE_TIME`, not `ISO_DATE_TIME` which doesn't include the offset (because we don't need it).

Do this, delete `targetUrlWithTime`, we should be good. 

Before you push, compile your code and run the tests on it (use `./run_all_tests.sh`) to make sure this is all okay (just want to make sure nothing else comes out of the woodwork) -- and it might be worth testing your code with the regression tests too.